### ISSUE TITLE
throw new ERR_INVALID_ARG_TYPE(name, 'string', value);

### ISSUE THREAD

**Issue by quarterioningot** [jantimon/html-webpack-plugin#1462]
## Expected behaviour
Webpack compiles and html is generatet

## Current behaviour
Webpack failes with error
![error](https://user-images.githubusercontent.com/8214583/84000415-e01c5b80-a964-11ea-8578-9dadc305b635.PNG)

## Environment
```
Node.js v11.10.0
win32 10.0.18363
npm 6.7.0

npm ls webpack
skeletons@1.0.0
`-- webpack@4.43.0

npm ls html-webpack-plugin
skeletons@1.0.0 
`-- html-webpack-plugin@4.3.0 
```

### Config
```js
const path = require("path");
const HtmlWebpackPlugin = require('html-webpack-plugin');

module.exports = {
  entry: "./src/index.ts",
  devtool: "inline-source-map",
  mode: "development",
  context: path.resolve(__dirname, 'dist'),
  optimization: {
    usedExports: true
  },
  module: {
    rules: [
      {
        test: /\.tsx?$/,
        use: "ts-loader",
        exclude: /node_modules/
      }
    ],
  },
  resolve: {
    extensions: [".tsx", ".ts", ".js"],
    plugins: [
      new HtmlWebpackPlugin({
        title: 'use plugin',
        filename: 'index.html'
      })
    ]
  },
  output: {
    filename: "main.js",
    path: path.resolve(__dirname, "dist")
  }
}
```

I have a pretty basic setup, with ts-loader and webpack. Not sure if I am missing something or an issue, but a simple 1-1 check from the examples what I have should work.

I debugged webpack and saw that `compiler.context` was undefined, and that is validating to be `undefined`. Maybe  this help.

![und-ctx](https://user-images.githubusercontent.com/8214583/84000867-cd565680-a965-11ea-8ac2-04dd077f4337.PNG)
![call-stack](https://user-images.githubusercontent.com/8214583/84000869-cfb8b080-a965-11ea-81bc-100c587ae9c9.PNG)

**Comment by jantimon** [comment]
Can it be fixed if you remove `context` from your configuration?

**Comment by quarterioningot** [comment]
> Can it be fixed if you remove `context` from your configuration?

I got the error regardless if ‘context’ is added or not. I even tried leaving the plugin as ‘  new HtmlWebpackPlugin()’

No luck

**Comment by jantimon** [comment]
I have never seen that `compiler.context` is empty

**Comment by cn-fanfare** [comment]
[https://github.com/jantimon/html-webpack-plugin/issues/1465](url)

i have same issue. 
why ROR in The "path" argument must be of type string. Received type object ?

**Comment by cn-fanfare** [comment]
how to fix this ?

**Comment by bryonsigawin** [comment]
I also had the same issue, even on a fresh project setup.
Managed to fix it after I noticed `new HtmlWebpackPlugin()` was placed in the wrong `plugin` field.

The offending line, placed under `resolve` > `plugins`.
```js
module.exports = {
  ...
  resolve: {
    plugins: [new HtmlWebpackPlugin()]
  },
  ...
}
```

It's supposed to be here, on the outer `plugin` field.
```js
module.exports = {
  ...
  plugins: [new HtmlWebpackPlugin()],
  ...
};
```

The [usage section](https://github.com/jantimon/html-webpack-plugin#usage) does show the correct placement too, which I've apparently missed 🤦‍♂️.
